### PR TITLE
TEP-0114: Migrate the testing Wait Task to reconcile on v1beta1.CustomRun

### PR DESCRIPTION
**Pull request by XinruZhang**
# Changes

As Custom Task Beta version is released, this PR creates a new Wait Task to reconcile on `v1beta1.CustomRun`, with the new condition changes applied.

/kind feature

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] ~Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing~
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] ~Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)~
- [x] ~Release notes contains the string "action required" if the change requires additional action from users switching to the new release~

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: test/custom-task-ctrls/wait-task-beta/pkg/reconciler/reconciler.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		r.Status.MarkCustomRunFailed("TimedOut", WaitTaskFailedOnRunTimeoutMsg)
+
+		return nil
+	}
+
+	// Don't emit events on nop-reconciliations, it causes scale problems.
+	return nil
+}
+
+func retryRun(r *v1beta1.CustomRun, c apis.Condition) {
```
Comment (jerop) [inline test/custom-task-ctrls/wait-task-beta/pkg/reconciler/reconciler.go:164]:
```suggestion
func retryCustomRun(r *v1beta1.CustomRun, c apis.Condition) {
```

**Review by jerop — COMMENTED** [review]
do you plan to add the [example run](https://github.com/tektoncd/experimental/blob/main/wait-task/example-run.yaml) in another PR? we could add it in this PR then add the [example pipeline](https://github.com/tektoncd/experimental/blob/main/wait-task/example-pipeline.yaml) in the next PR when it's supported in pipeline controller

**Comment by XinruZhang** [comment]
> do you plan to add the [example run](https://github.com/tektoncd/experimental/blob/main/wait-task/example-run.yaml) in another PR? we could add it in this PR then add the [example pipeline](https://github.com/tektoncd/experimental/blob/main/wait-task/example-pipeline.yaml) in the next PR when it's supported in pipeline controller

Thanks @jerop, added it in this PR!

**Review by jerop — APPROVED** [review]
Thank you @XinruZhang!

**Review by dibyom — COMMENTED** [review]
/lgtm

**Pull request merged**